### ISSUE TITLE
Number of PCA points larger than number of correspondence points

### ISSUE THREAD

**Issue by ricardopizarro** [SCIInstitute/ShapeWorks#2283]
The number of PCA components that is rendered in the analysis portion of the software is always the same 224 components. I have tried changing the number of correspondence points to [64, 246, 1024] and it always generates 224 components. I am not sure how we can have more PCA components than correspondence points. Can you please provide some insight?

**Comment by akenmorris** [comment]
You're right that we should limit the number of components to the number of dimensions of the data (typically 3x the number of particles, for x, y and z).  You should see that all of the ones past the number of dimensions will be nearly zero.

You probably have 225 samples in your project, which also limits the number of PCA modes.  The data is fully described in N-1 modes of variation and the usual case for ShapeWorks is that the number of dimensions far exceed the number of samples.  E.g 300 shapes with 128 particles is constrained by the number of samples (300) not the number of dimensions (384).
